spot_driver/spot_driver/spot_ros2.py

```diff
@@ -738,6 +740,17 @@ def bodyPoseCallback(self, data):
         mobility_params.body_control.CopyFrom(body_control)
         self.spot_wrapper.set_mobility_params(mobility_params)
 
+    def handle_graph_nav_upload_graph(self, request, response):
+        try:
+            self.node.get_logger().info(f"Uploading GraphNav map: {request.upload_filepath}")
+            self.spot_wrapper._upload_graph_and_snapshots(request.upload_filepath)
+            self.node.get_logger().info(f"Uploaded")
+            response.success = True
+            response.message = "Success"
+        except Exception as e:
```

Comment:
I tried to find but did not succeed in understanding the value of `response` coming into this function. I see other callbacks set `response.success = False` explicitly, and some do not.

```
    def handle_stair_mode(self, request, response):
        """ROS service handler to set a stair mode to the robot."""
        try:
            mobility_params = self.spot_wrapper.get_mobility_params()
            mobility_params.stair_hint = request.data
            self.spot_wrapper.set_mobility_params(mobility_params)
            response.success = True
            response.message = 'Success'
            return response
        except Exception as e:
            response.success = False
            response.message = 'Error:{}'.format(e)
            return response
       ```
       (the `return` can be hoisted out)

Reply:
right, I don't understand in ROS 2 what `response` is set to on entry to the function. e.g. does it default to `success = False`.